### PR TITLE
refactor(treesitter): remove duplicated diagnostic code

### PR DESCRIPTION
**Pull request by MariaSolOs**
Updates some of the error messages to include the improvements from #24393, and removes the code that used to generate specific diagnostics since those are now coming from `treesitter.c`.

### CODE REVIEW

**Comment by MariaSolOs** [comment]
@clason lmk if this is what you had in mind ;)

**Comment by clason** [comment]
Now _that's_ a proper diffstat ;)

**Comment by clason** [comment]
Nice! Would it be possible to strip the leading "Query error at x:y. "? That's kinda redundant with the virtual text, and eats valuable space.

The `x:y` could also be used to put the virtual text in the exact position for multi-line patterns.

**Comment by clason** [comment]
@lewis6991

**Comment by MariaSolOs** [comment]
@clason I can remove the prefix.

> The x:y could also be used to put the virtual text in the exact position for multi-line patterns.

Won’t the virtual text always be appended at the end of the line?

**Comment by clason** [comment]
> Won’t the virtual text always be appended at the end of the line?

Yes, but the first line rather than the line `x`. Here's what I mean:
Screenshot 2023-09-12 at 17 30 45

The actual error is in the second line (column 10), but the virtual text is in the first line of the range.

**Comment by MariaSolOs** [comment]
@clason I agree with removing the information after the colon, but I do find the position information helpful. What do you think of `2:10: Invalid capture name "onstant.builtin"`?

As for modifying the range, that might be trickier since right now we just [use the range of the problematic node](https://github.com/neovim/neovim/blob/4d1fed3ac841a5d0ae99a157723fd537c7b3a1b0/runtime/lua/vim/treesitter/_query_linter.lua#L32). I guess I can add extra stuff to better narrow the range...

**Comment by clason** [comment]
> but I do find the position information helpful. 

Interesting. How so? Isn't it redundant with the actual thing right next to it?

> As for modifying the range, that might be trickier since right now we just [use the range of the problematic node](https://github.com/neovim/neovim/blob/4d1fed3ac841a5d0ae99a157723fd537c7b3a1b0/runtime/lua/vim/treesitter/_query_linter.lua#L32). I guess I can add extra stuff to better narrow the range...

Ah, it seemed to me like it just marked the whole pattern instead of the affected node. But this is not a biggie; was just a "nice to have" thought.

**Comment by MariaSolOs** [comment]
> Interesting. How so? Isn't it redundant with the actual thing right next to it?

Right but given the limited space having the exact position (including the column number) right at the beginning of the error is helpful.

**Comment by lewis6991** [comment]
Is that what diagnostic highlights are for?

**Comment by MariaSolOs** [comment]
@lewis6991 are you referring to my position comment? I don't feel to strongly about it BTW, I'm fine with removing it.

**Comment by lewis6991** [comment]
> right at the beginning of the error is helpful. With all the parentheses it's easy to get lost in these queries :D

I'm not sure how you can get lost since diagnostics have highlights. Also this position information should be stored in the diagnostic so if you want this information in the virtual text then all you need is a custom virt text handler to display it.

**Comment by MariaSolOs** [comment]
> I'm not sure how you can get lost since diagnostics have highlights.

@lewis6991 right, but given that the query editor highlights the entire node instead of just the problematic bit (e.g. the invalid capture name), I find the position helpful.

> Also this position information should be stored in the diagnostic so if you want this information in the virtual text then all you need is a custom virt text handler to display it.

This doesn't bother me enough to write a custom virtual text handler for the query editor.

But again, I'm fine with it. I'll remove it and go with @clason's original suggestion.

**Comment by lewis6991** [comment]
> @lewis6991 right, but given that the query editor highlights the entire node instead of just the problematic bit (e.g. the invalid capture name), I find the position helpful.

That sounds like something we should fix. I kinda thought that was the point of this PR.

**Comment by MariaSolOs** [comment]
@lewis6991 the point of this PR was to update the diagnostics from the query linter to take advantage of the changes from #24393.

**Comment by clason** [comment]
>  but given that the query editor highlights the entire node instead of just the problematic bit (e.g. the invalid capture name), I find the position helpful.

Yes; to be clear, I was suggesting to change that.

**Comment by lewis6991** [comment]
> @lewis6991 the point of this PR was to update the diagnostics from the query linter to take advantage of the changes from #24393.

#24393 provides more information on the location of the error, we should take advantage of that.

**Comment by MariaSolOs** [comment]
> Yes; to be clear, I was suggesting to change that.

@clason My bad, I thought from your comment below that we were fine with highlighting the entire node.
> Ah, it seemed to me like it just marked the whole pattern instead of the affected node. But this is not a biggie; was just a "nice to have" thought.

**Comment by lewis6991** [comment]
It's pretty suboptimal to have a diagnostic at position `X` to contain the message `error at position Y`. 

It's not just highlighting that diagnostics feed into, it's also signs, navigation quickfix, etc.

**Comment by clason** [comment]
Sorry, that was my bad. That comment was meant to convey that I will (grudgingly) accept highlighting of the entire pattern, but would prefer a more targeted highlight.

**Comment by MariaSolOs** [comment]
@clason @lewis6991 I will modify the diagnostic range. Sorry for misunderstanding your previous suggestions.

**Comment by MariaSolOs** [comment]
Okay this is how it's looking now:
image

File: runtime/lua/vim/treesitter/_query_linter.lua [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
--- those node names exist for every language
-local BUILT_IN_NODE_NAMES = { '_', 'ERROR' }
 
 local M = {}
 
 --- @class QueryLinterNormalizedOpts
 --- @field langs string[]
 --- @field clear boolean
 
+--- @alias ParseError {msg: string, range: Range4}
```
Comment (lewis6991) [inline runtime/lua/vim/treesitter/_query_linter.lua:11]: Make sure to namespace types appropriately. Use `vim.` at minimum.

Comment (lewis6991) [reply runtime/lua/vim/treesitter/_query_linter.lua:11]: It's a shame LuaLs doesn't have real namespacing yet. The verbose type names can be a little annoying for simple aliases.

**Comment by clason** [comment]
LGTM! @lewis6991?

**Pull request merged**